api/src/main/java/uk/gov/dvsa/mot/app/MotrRequestHandler.java

```diff
@@ -0,0 +1,199 @@
+import uk.gov.dvsa.mot.trade.read.core.MotrReadService;
+import uk.gov.dvsa.mot.trade.service.AnnualTestExpiryDateCalculator;
+import uk.gov.dvsa.mot.vehicle.hgv.HgvVehicleProvider;
+import uk.gov.dvsa.mot.vehicle.hgv.model.HgvPsvVehicle;
+import uk.gov.dvsa.mot.vehicle.hgv.model.TestHistory;
+import uk.gov.dvsa.mot.vehicle.hgv.model.Vehicle;
+
+import java.util.Optional;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.Response;
+
+@Path("/")
+public class MotrRequestHandler extends AbstractRequestHandler {
+    private static final Logger logger = Logger.getLogger(MotrRequestHandler.class);
+
+    private MotrReadService motrReadService;
+    private HgvVehicleProvider hgvVehicleProvider;
+    private String awsRequestId;
+
+    public MotrRequestHandler() {
+        super();
+    }
+
+    public MotrRequestHandler(boolean injectSelf) {
+        super(injectSelf);
+    }
+
+    @Inject
+    public void setHgvVehicleProvider(HgvVehicleProvider hgvVehicleProvider) {
+        this.hgvVehicleProvider = hgvVehicleProvider;
+    }
+
+    @Inject
+    public void setMotrReadService(MotrReadService motrReadService) {
+        this.motrReadService = motrReadService;
+    }
+
+    @GET
+    @Path("motr/v2/search/registration/{registration}")
+    @Produces("application/json")
+    public Response getVehicle(@PathParam("registration") String registration,
+                               ContainerRequestContext requestContext) throws TradeException {
+        try {
+
+            ApiGatewayRequestContext context = (ApiGatewayRequestContext) requestContext.getProperty(
+                    RequestReader.API_GATEWAY_CONTEXT_PROPERTY);
+            if (context != null) {
+                awsRequestId = context.getRequestId();
+            }
+
+            logger.info(String.format("Entering MotrRequestHandler, awsRequestId: %s", awsRequestId));
+
+            Optional<MotrResponse> motVehiclesOptional = getMotVehicle(registration);
+
+            if (motVehiclesOptional.isPresent()) {
+
+                return Response.ok(buildMotResponse(motVehiclesOptional.get())).build();
+            }
+
+            Optional<MotrResponse> dvlaVehicleOptional = getDvlaVehicle(registration);
+
+            if (dvlaVehicleOptional.isPresent()) {
+                MotrResponse dvlaVehicle = dvlaVehicleOptional.get();
+
+                Optional<Vehicle> hgvPsvVehicleOptional = getHgvPsvVehicle(registration);
+
+                if (!hgvPsvVehicleOptional.isPresent()) {
+                    logger.info("No HGV/PSV vehicle retrieved");
+
+                    if (dvlaVehicle.getMotTestExpiryDate() != null) {
+                        return Response.ok(buildMotResponse(dvlaVehicle)).build();
+                    }
+
+                    throw new InvalidResourceException("No HGV/PSV vehicle retrieved for registration " + registration,
```

Comment:
what is InvalidResourceException ? Does it end up with 404 ?

Reply:
Aye, its 404 with error type "Invalid Resource".